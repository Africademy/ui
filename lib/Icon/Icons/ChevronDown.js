"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _createIconComponent = _interopRequireDefault(require("../utils/createIconComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
var ChevronDownIcon = (0, _createIconComponent.default)({
  content: _react.default.createElement("g", null, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11.146 6.854A.5.5 0 0 0 10.793 6H5.207a.5.5 0 0 0-.353.854l2.792 2.792a.5.5 0 0 0 .708 0l2.792-2.792z"
  }))
});
ChevronDownIcon.displayName = 'ChevronDownIcon';
var _default = ChevronDownIcon;
exports.default = _default;