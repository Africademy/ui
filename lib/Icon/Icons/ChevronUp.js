"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _createIconComponent = _interopRequireDefault(require("../utils/createIconComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
var ChevronUpIcon = (0, _createIconComponent.default)({
  content: _react.default.createElement("g", null, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11.146 9.146a.5.5 0 0 1-.353.854H5.207a.5.5 0 0 1-.353-.854l2.792-2.792a.5.5 0 0 1 .708 0l2.792 2.792z"
  }))
});
ChevronUpIcon.displayName = 'ChevronUpIcon';
var _default = ChevronUpIcon;
exports.default = _default;