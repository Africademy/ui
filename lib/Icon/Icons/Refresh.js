"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _createIconComponent = _interopRequireDefault(require("../utils/createIconComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
var RefreshIcon = (0, _createIconComponent.default)({
  content: _react.default.createElement("g", null, _react.default.createElement("path", {
    d: "M7 1a7 7 0 1 0 0 14h1a1 1 0 1 0 0-2H7a5 5 0 1 1 5-5h-1.793a.5.5 0 0 0-.353.854l2.793 2.793a.5.5 0 0 0 .707 0l2.793-2.793A.5.5 0 0 0 15.793 8H14a7 7 0 0 0-7-7z"
  }))
});
RefreshIcon.displayName = 'RefreshIcon';
var _default = RefreshIcon;
exports.default = _default;