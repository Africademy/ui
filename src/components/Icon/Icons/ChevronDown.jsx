import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';
import { StyledIcon } from './style';

const ChevronDown = ({ type, size }) => (
  <Icon type={type} size={size}>
    <StyledIcon type={type} fillRule="evenodd" clipRule="evenodd" d="M7.14645 0.853553C7.46143 0.53857 7.23835 0 6.79289 0L1.20711 0C0.761654 0 0.538571 0.538571 0.853553 0.853553L3.64645 3.64645C3.84171 3.84171 4.15829 3.84171 4.35355 3.64645L7.14645 0.853553Z" fill="#0F0B0A" />
  </Icon>
);

ChevronDown.propTypes = {
  size: PropTypes.oneOf('small', 'medium', 'large'),
  type: PropTypes.oneOf('primary', 'secondary'),
};

ChevronDown.defaultProps = {
  size: 'medium',
  type: 'secondary',
};


export default ChevronDown;
