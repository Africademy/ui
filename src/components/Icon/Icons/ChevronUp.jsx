import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';
import { StyledIcon } from './style';

const ChevronUp = ({ type, size }) => (
  <Icon type={type} size={size}>
    <StyledIcon type={type} fillRule="evenodd" clipRule="evenodd" d="M7.14645 3.14645C7.46143 3.46143 7.23835 4 6.79289 4L1.20711 4C0.761654 4 0.538571 3.46143 0.853553 3.14645L3.64645 0.353553C3.84171 0.158291 4.15829 0.158291 4.35355 0.353553L7.14645 3.14645Z" fill="#0F0B0A" />
  </Icon>
);

ChevronUp.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  type: PropTypes.oneOf(['primary', 'secondary']),
};

ChevronUp.defaultProps = {
  size: 'medium',
  type: 'secondary',
};

export default ChevronUp;
