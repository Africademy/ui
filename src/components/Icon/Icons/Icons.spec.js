import snap from 'jest-auto-snapshots';
import ChevronUp from './ChevronUp';
import ChevronDown from './ChevronDown';
import Refresh from './Refresh';

snap(ChevronUp, './ChevronUp.jsx');
snap(ChevronDown, './ChevronDown.jsx');
snap(Refresh, './Refresh.jsx');
