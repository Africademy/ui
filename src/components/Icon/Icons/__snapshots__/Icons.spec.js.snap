// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > ChevronDown Matches snapshot when passed all props 1`] = `
<Icon
  size="small"
  type="primary"
>
  <ForwardRef
    clipRule="evenodd"
    d="M7.14645 0.853553C7.46143 0.53857 7.23835 0 6.79289 0L1.20711 0C0.761654 0 0.538571 0.538571 0.853553 0.853553L3.64645 3.64645C3.84171 3.84171 4.15829 3.84171 4.35355 3.64645L7.14645 0.853553Z"
    fill="#0F0B0A"
    fillRule="evenodd"
    type="primary"
  />
</Icon>
`;

exports[`jest-auto-snapshots > ChevronUp Matches snapshot when passed all props 1`] = `
<Icon
  size="small"
  type="primary"
>
  <ForwardRef
    clipRule="evenodd"
    d="M7.14645 3.14645C7.46143 3.46143 7.23835 4 6.79289 4L1.20711 4C0.761654 4 0.538571 3.46143 0.853553 3.14645L3.64645 0.353553C3.84171 0.158291 4.15829 0.158291 4.35355 0.353553L7.14645 3.14645Z"
    fill="#0F0B0A"
    fillRule="evenodd"
    type="primary"
  />
</Icon>
`;

exports[`jest-auto-snapshots > RefreshIcon Matches snapshot when passed all props 1`] = `
<Icon
  color="primary"
  size="small"
  type="secondary"
>
  <ForwardRef
    d="M7 0C3.134 0 0 3.134 0 7C0 10.866 3.134 14 7 14H8C8.55228 14 9 13.5523 9 13C9 12.4477 8.55228 12 8 12H7C4.23859 12 2 9.76141 2 7C2 4.23859 4.23859 2 7 2C9.76141 2 12 4.23859 12 7H10.2071C9.76166 7 9.53857 7.53857 9.85355 7.85355L12.6465 10.6465C12.8417 10.8417 13.1583 10.8417 13.3535 10.6465L16.1465 7.85355C16.4614 7.53857 16.2383 7 15.7929 7H14C14 3.134 10.866 0 7 0Z"
    fill="#0F0B0A"
    type="primary"
  />
</Icon>
`;
